consensus/src/aggregator.rs

```diff
@@ -75,29 +71,69 @@ impl QuorumMaker {
     }
 
     /// Try to append a signature to a (partial) quorum.
-    pub fn append(&mut self, vote: Vote, committee: &Committee) -> ConsensusResult<Option<Votes>> {
+    pub fn append(&mut self, vote: Vote, committee: &Committee) -> ConsensusResult<Option<QC>> {
         let author = vote.author;
 
-        // Ensure the authority has voting rights.
-        let voting_rights = committee.stake(&author);
-        ensure!(voting_rights > 0, ConsensusError::UnknownAuthority(author));
-
         // Ensure it is the first time this authority votes.
         ensure!(
             self.used.insert(author),
             ConsensusError::AuthorityReuse(author)
         );
 
-        // Check the signature on the vote.
-        vote.signature.verify(&vote.digest(), &author)?;
-
         self.votes.push((author, vote.signature));
-        self.weight += voting_rights;
+        self.weight += committee.stake(&author);
+        if self.weight >= committee.quorum_threshold() {
+            self.weight = 0; // Ensures QC is only made once.
+            return Ok(Some(QC {
+                hash: vote.hash.clone(),
+                round: vote.round,
+                votes: self.votes.clone(),
+            }));
+        }
+        Ok(None)
+    }
+}
+
+struct TCMaker {
```

Comment:
Do we need the TCMaker in 3-chain?

Reply:
no, there are many things we don't need. I am rebasing it now